### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by c00p75**
# Blog App: validations and model specs

## Objectives
- [ ] **Validations**
- Added validations for User model:
  > - Name must not be blank.
  > - PostsCounter must be an integer greater than or equal to zero.

- Added validations for Post model:
  > - Title must not be blank.
  > - Title must not exceed 250 characters.
  > - CommentsCounter must be an integer greater than or equal to zero.
  > - LikesCounter must be an integer greater than or equal to zero.

- [ ] **Unit tests**
- Added unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  describe 'scopes' do
+    it 'recent posts count should be 3' do
+      Post.create(author: @user, title: 'post 1', text: 'content', comments_counter: 3, likes_counter: 3)
+      Post.create(author: @user, title: 'post 2', text: 'content', comments_counter: 3, likes_counter: 3)
+      Post.create(author: @user, title: 'post 3', text: 'content', comments_counter: 3, likes_counter: 3)
+      Post.create(author: @user, title: 'post 3', text: 'content', comments_counter: 3, likes_counter: 3)
+      expect(User.recent_posts(@user).count).to eq(3)
+    end
```
Comment (BenMukebo) [inline spec/models/user_spec.rb:32]: - [x] According to the project requirements, Any custom methods in the models should have a corresponding unit test. Kindly make sure the `update_posts_counter` and `recent_comments` methods from the Post model have a corresponding unit test. ✍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @c00p75,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Gitflow was correctly used.  ✔️   
- Professional README ✔️ 
- All the tests passed. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 1 above)
Comment (c00p75) [reply spec/models/user_spec.rb:32]: Thanks for the feedback @BenMukebo.

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @c00p75 

Your project is complete! Great job for adding the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- unit test for methods well added
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**